### ISSUE TITLE
Feature Request - Ability to add a label to `<SearchBox />`

### ISSUE THREAD

**Issue by Bouncey** [algolia/react-instantsearch#1087]
Hey guys, great module you have here. Makes working with the REST API a dream. 

I couldn't find any issues or forum topics on this. I have a feeling I wasn't looking in the right places. 

## Feature Request - Ability to add a label to `<SearchBox />`

**Feature: What is your use case for such a feature?**
A11y

**Feature: What is your proposed API entry? The new option to add? What is the behavior?**
```jsx 
// consumer API
<SearchBox 
  labelledBy={ LabelComponent }
/>

// packages/react-instantsearch/src/components/SearchBox.js
class SearchBox extends Component {
  // [...]
  render() {
    const {
      // ...
      labelledBy: Label
      // ...
  } = this.props;
  const Input = () => <input { ...inputProps } />;

  return (
    // [...]
    {
      Label ? <Label><Input /></Label> : <Input />
    }
    // [...]
  );
```

**What is the version you are using? Always use the latest one before opening a bug issue.**
V5.0.1

**Comment by Haroenv** [comment]
Good idea. For now, you’ll be able to copy the component we have for searchBox and add the label, but it should be there by default (no visible text needed though IMO)
